mxd/performance-tests/mxd-performance-evaluation/results_aggregation.py

```diff
@@ -1,85 +1,222 @@
-
-        # Adds labels to each point on the scatter plot (that x-axis matches OEM_PLANTS values)
-        for i, text in enumerate(labels):
-            plt.text(stats[i][0], stats[i][1], text)
-
-        # Add cursor
-        cursor = mplcursors.cursor(scatter, hover=True)
-        @cursor.connect("add")
-        def on_add(sel):
-            i = sel.target.index
-            sel.annotation.set_text('Plants: {}, Cars: {}, Parts/Car: {}, Cars/Interval: {}'.format(
-                stats[i][0],
-                meta_values_list[i]['OEM_CARS_INITIAL'],
-                meta_values_list[i]['PARTS_PER_CAR'],
-                meta_values_list[i]['CARS_PRODUCED_PER_INTERVALL']))
-
-        plt.xlabel('OEM_PLANTS')
-        plt.ylabel(value)
-        plt.show()
\ No newline at end of file
+import sys
+import numpy as np
+from sklearn.linear_model import LinearRegression
+import argparse
+
+OEM_CARS_INITIAL = 'OEM_CARS_INITIAL'
+SUPPLIER_PARTS_INITIAL = 'SUPPLIER_PARTS_INITIAL'
+OEM_PLANTS = 'OEM_PLANTS'
+SUPPLIER_PLANTS = 'SUPPLIER_PLANTS'
+SUPPLIER_FLEET_MANAGERS = 'SUPPLIER_FLEET_MANAGERS'
+ADDITIONAL_CONTRACT_DEFINITIONS_OEM = 'ADDITIONAL_CONTRACT_DEFINITIONS_OEM'
+ADDITIONAL_CONTRACT_DEFINITIONS_SUPPLIER = 'ADDITIONAL_CONTRACT_DEFINITIONS_SUPPLIER'
+DEFAULT_OUTPUT_FILE = 'output.html'
+USAGE_MESSAGE = 'Usage: python3 results_aggregation.py <directory> <html_file> --regression <operation_name>'
+
+
+def load_metadata(metadata_file):
+    """
+    Load metadata from a text file.
+    """
+    metadata = {}
+
+    with open(metadata_file, 'r') as f:
+        for line in f:
+            line = line.strip()
+            if "=" in line:
+                key, value = line.split('=')
+                metadata[key.strip()] = value.strip()
+    return metadata
+
+
+def load_stats(stat_file):
+    """
+    Load statistics from a JSON file.
+    """
+    with open(stat_file, 'r') as file:
+        data = json.load(file)
+
+    processes_data = {}
+
+    for process_name, process_data in data.items():
+        median_res_time = process_data['medianResTime']
+        processes_data[process_name] = median_res_time
+
+    return processes_data
+
+
+def sort_processes(processes, metadata):
+    """
+    Sort processes based on specified criteria.
+    """
+    sorted_processes = sorted(processes, key=lambda x: (
+        int(metadata[x].get(OEM_CARS_INITIAL)),
+        int(metadata[x].get(SUPPLIER_PARTS_INITIAL)),
+        int(metadata[x].get(OEM_PLANTS)),
+        int(metadata[x].get(SUPPLIER_PLANTS)),
+        int(metadata[x].get(SUPPLIER_FLEET_MANAGERS)),
+        int(metadata[x].get(ADDITIONAL_CONTRACT_DEFINITIONS_OEM)),
+        int(metadata[x].get(ADDITIONAL_CONTRACT_DEFINITIONS_SUPPLIER)),
+    ))
+    return sorted_processes
+
+
+def erase_file_contents(filename):
+    if os.path.exists(filename):
+        with open(filename, 'w') as file:
+            pass
+
+
+def plot_process_data(scenarios, stats_data, output_file):
```

Comment:
Response times are provided as ms. This fact is only explicitly stated inside the dashboard.js embedded in the JMeter HTML output. Please also specify (ms) on the y-axis of the corresponding plots.